File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ind = 0, cnt =0;

        for(int i=0; i<nums.size(); i++)
        {
            if(nums[i] == nums[ind])
            {
                cnt++;
            }
            else if(nums[i] != nums[ind] && cnt != 0) 
            {
                cnt--;
            }
            else 
            {
                ind = i;
                cnt++;
            }
        }

        return nums[ind];
    }
};

File: 169-majority-element/majority-element.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();

        for (int ele : nums) {
            mp.put(ele, mp.getOrDefault(ele, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : mp.entrySet()) {
            if (entry.getValue() > nums.length / 2) {
                return entry.getKey();
            }
        }

        return -1;
    }
}
